### ISSUE TITLE
Printing of `Expr(:thunk)`

### ISSUE THREAD

**Issue by yuyichao** [JuliaLang/julia#18182]
Right now the printing of `Expr(:thunk)` is basically useless

``` jl
julia> expand(:(for i in a end))
:($(Expr(:thunk, Toplevel LambdaInfo thunk)))
```

**Comment by vtjnash** [comment]
I think we should always print the full expansion of the AST instead of `Toplevel LambdaInfo thunk`

**Comment by ihnorton** [comment]
Seems fixed (#18413 and related)
```
julia> expand(Main, :(for i in a end))
:($(Expr(:thunk, CodeInfo(:(begin
        SSAValue(0) = a
        #temp# = (Base.start)(SSAValue(0))
        3:
        unless !((Base.done)(SSAValue(0), #temp#)) goto 13
        SSAValue(1) = (Base.next)(SSAValue(0), #temp#)
        i = (Core.getfield)(SSAValue(1), 1)
        #temp# = (Core.getfield)(SSAValue(1), 2)
        # meta: location REPL[2]
        #= REPL[2]:1 =#
        # meta: pop location
        11:
        goto 3
        13:
        return
    end)))))
```
